project/Yu-Rou-Weng/proposal.rst

```diff
@@ -0,0 +1,169 @@
+to serve as a useful plugged-in library for easier automation.
+
+Prospective Users
+=================
+
+.. Describe the users of your software and how they will use it.  It is OK to
+.. combine this section with the previous one (`Problem to solve`_).
+
+1.Modmesh: (Will add more explanation after discussions)
+
+2.Fluid dynamics professionals or researchers who need to exchange mesh data between Plot3D and SU2 when 
+using different simulation software.
+
+3.In the aerospace and automotive industries, engineers frequently rely on computational fluid dynamics (CFD) 
+simulations. This converter offers a more precise mesh data conversion solution for their simulations.
+
+4.Computer Scientists, Students, and educators who are working in computational fluid dynamics studies and experiments.
+
+System Architecture
+===================
+(Will finish after discussion with yy teacher)
+
+.. Analyze how your system takes input, produces results, provide interface, and
+.. erforms any other operations.  Describe the system's work flow.  You may
+.. consider to use a flow chart but it is not required.  Specify the constraints
+.. assumed in your system.  Describe the modularization of the system.
+
+API Description
+===============
+(Still in progress, will finish after discussion with yy teacher)
+
+1.Fuctions in c++ will include ReadPlot3DData、WriteSU2Data、ConvertPlot3DToSU2
+
+2.Use pybind11 for wrapping
+
+.. Show how your system can be programmed.  You are supposed to implement the
+.. system using both C++ and Python.  Describe how a user writes a script in the
+.. system.
+
+Engineering Infrastructure
+==========================
+
+.. Describe how you plan to put together the engineering system:
+
+.. 1. Automatic build system and how to build your program
+.. 2. Version control (show how you will use it)
+.. 3. Testing framework
+.. 4. Documentation
+
+.. Some of the above information should be included in the documentation in your
+.. software.
+
+.. You may use continuous integration, but it is not required.  If you use it,
+.. describe how it works in your code development.
+
+1. Automatic build system:**CMake**
+2. Version control: Git
+3. Testing framework: **Pytest** 
+4. Documentation: README(or Notion)、 [PyDoc](https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring)
+
+Schedule
+========
+
+.. Itemize the work to do and list them in a timeline.  Estimate the efforts of
+.. each item before the project starts.
+
+.. The schedule is expected to be adjusted during the development.  A schedule to
+.. accurately predict everything in the development is impossible and unnecessary.
+.. But the initial estimate is important, for the baseline becomes concrete when
+.. we adjust the plan.
+
+.. To start, you can list the to-do items on a weekly basis and assume the
+.. development of the project takes 8 weeks:
+
+* Planning phase (8 weeks from mm/dd to mm/dd):
+* Week 1 (10/30):
```

Comment:
It will be more productive to start with a prototype, and use the prototyping work to study the domain related.  That is, when studying, write some code for it.